plaidml/edsl/edsl.h

```diff
@@ -1081,7 +1084,21 @@ inline Tensor zero() { return Tensor{0}; }
 inline Program::Program(                 //
     const std::string& name,             //
     const std::vector<Tensor>& outputs,  //
+    const DType& floatx,                 //
+    const DType& intx,                   //
     const std::vector<std::tuple<Tensor, Tensor>>& updates) {
+  if (!isFloat(floatx)) {
```

Comment:
Let's have all this validation occur inside the `ffi.cc` layer instead of out in the user's context.